api/v1beta1/hostedcluster_types.go

```diff
@@ -433,7 +433,10 @@ type HostedClusterSpec struct {
 }
 
 // OLMCatalogPlacement is an enum specifying the placement of OLM catalog components.
+//
 // +kubebuilder:validation:Enum=management;guest
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="OLMCatalogPlacement is immutable"
```

Comment:
why is this here instead of in https://github.com/openshift/hypershift/blob/main/api/v1beta1/hostedcluster_types.go#L424-L427